src/prepare.js

```diff
@@ -43,14 +49,19 @@ function createCompositeElementInstance(
 }
 
 function renderCompositeElementInstance(instance, context = {}) {
-  const childContext = Object.assign(
-    {},
-    context,
-    instance.getChildContext ? instance.getChildContext() : {},
-  );
+  const childContext = {
+    ...context,
+    ...(instance.getChildContext ? instance.getChildContext() : {}),
+  };
   return [instance.render(), childContext];
 }
 
+function renderFunctionElementInstance({ type, props }, context = {}) {
+  ReactSharedInternals.ReactCurrentDispatcher.current =
+    createDispatcher(context);
+  return type(props);
```

Comment:
Why do you set this for every function component we render? I don't see why it wouldn't be fine to just set it once and reuse the same dispatcher?

Reply:
We don't need to set it for every component. It was just easier that way. The `prepare`-function is run recursively so setting it there will set it for every render of _all_ components. 
I suppose I could create a setup function that just sets up the dispatcher, then calls `prepare`.

Reply:
I think a quick `if` in the `prepare` function should do it :). I just imagine this creating quite a bit of overhead when setting up and assigning 10 functions for say 100 components, instead of once (and creating extra work for the GC).

Reply:
Right 🤦, of course, that would be easier. 
However, I believe it might also be a good idea to set the dispatcher back to what it used to be after we finish using it. Probably not strictly necessary but I noticed `react-shallow-renderer` does this.

Reply:
Yeah, you could I guess, but since react-prepare is not meant to be used in-between renders, it shouldn't be an issue. ReactDOMs `renderToString` should take care of this. So it's not strictly necessary.

Reply:
Ok, so it turns out we do need to change the dispatcher between each render because of the context. But I [changed it](https://github.com/webkom/react-prepare/blob/fb1cd1485014000bd2d2c38e6e8ea2cd495c7149/src/utils/createDispatcher.js) so that all functions are defined only once, but we create a new dispatcher that references the functions on each render. I believe this should be a lot more performant.

Reply:
Yeah, that's a lot better. The other idea I though of, is just store the "current" context in some global state the dispatcher has access to. This way you can just set the context somewhere, like `SOME_GLOBAL_VAR.current_context = context`.
I would say that it's slightly cleaner, and saves on the object constructions of the dispatcher, but I think it's fine if you think the current implementation is better.

Reply:
[New version](https://github.com/webkom/react-prepare/blob/04d1e485f70877a3b95cdd7d3af1594014c27c8f/src/utils/createDispatcher.js). I did something like that. Now the whole dispatcher is a global variable, so the context can be contained within the dispatcher. I also added some functions to it, to make the code in `prepare` prettier :)